crates/sui-protocol-config/src/lib.rs

```diff
@@ -1363,6 +1372,12 @@ impl ProtocolConfig {
 
                 cfg
             }
+            21 => {
+                let mut cfg = Self::get_for_version_impl(version - 1, chain);
+                cfg.feature_flags.shared_object_deletion = true;
```

Comment:
i think we will probably want to test this on devnet before releasing to testnet/mainnet due to the potential impact of any bugs. See version 12 for an example of how we enabled zklogin on devnet only.